dashboard/app/controllers/featured_projects_controller.rb

```diff
@@ -1,23 +1,19 @@
 class FeaturedProjectsController < ApplicationController
-  def create
-    unless current_user.try(:permission?, UserPermission::PROJECT_VALIDATOR)
-      _, channel_id = storage_decrypt_channel_id(featured_project_params[:project_id])
-      @featured_project = FeaturedProject.create({storage_app_id: channel_id, who_featured_user_id: current_user.id})
-      @featured_project.save!
-    end
-  end
+  authorize_resource
 
-  def destroy_by_project_id
-    unless current_user.try(:permission?, UserPermission::PROJECT_VALIDATOR)
-      _, channel_id = storage_decrypt_channel_id(params[:project_id])
-      @featured_project = FeaturedProject.find_by_storage_app_id(channel_id)
-      @featured_project.destroy
-    end
+  def feature
+    _, channel_id = storage_decrypt_channel_id(params[:project_id])
+    return render_404 unless channel_id
+    @featured_project = FeaturedProject.find_or_create_by(storage_app_id: channel_id)
```

Comment:
We don't expect this to fail, but to be safe use the bang version [find_or_create_by!](https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by%21) in case the create fails. Either way we'll get a honeybadger error but the creation error will be more helpful than trying to call `update_attribute` on `nil` on the next line.

Reply:
✅